euphonic/qpoint_phonon_modes.py

```diff
@@ -594,6 +594,88 @@ def calculate_pdos(
 
         return Spectrum1DCollection(
             dos_bins, all_dos_y_data, metadata=metadata)
+    
+    def calculate_displacement(self, qpt_index: int, mode: Union[int, np.ndarray], disp_amplitude: np.complex128, supercell: np.ndarray,) -> Crystal:
+        """
+        Determine the displacement of each atom in the supercell
+        according to the eigenmode with wavevector :math:`\\mathbf{Q}`
+        and index :math:`\\nu`. According to [1]_ displacement of atom
+        indexed :math:`\\kappa` is:
+
+        .. math::
+
+            \\mathbf{u}(\\kappa,\\mathbf{q},\\nu) = 
+                A
+                \\frac{1}{\\sqrt{M_\\kappa}}
+                \\mathbf{e}_{\\mathbf{q}\\nu\\kappa}
+                \\exp\\left( i\\mathbf{Q}{\\cdot}\\mathbf{r}_{\\kappa} \\right)
+                
+
+                
+
+        Where A is the displacement amplitude, :math:`M_{\\kappa}` is the atom mass,
+        :math:`\\mathbf{e}_{\\mathbf{q}\\nu\\kappa}` is the displacement 
+        of atom :math:`\\kappa` according to the eigenvector of phonon
+        with wavevector :math:`\\mathbf{q}` and band index :math:`\\nu`.
+
+        .. [1] M.T. Dove, Structure and Dynamics, Oxford University Press, Oxford, 2003, pp. 211-212.
+
+
+        Parameters
+        ----------
+        qpt_index
+            Index of the Q vector.
+        mode
+            Index of the mode, or array of amplitudes for each mode.
+            Providing an array allows mixing degenerated modes.
```

Comment:
This double-duty feels a bit awkward. Can you use complex numbers here as well? Is it useful to set disp_amplitude if also passing amplitudes here? Is it permitted to combine modes from multiple Q vectors?
It's helpful to minimise the number of paths through a function if possible; it makes the overall system easier to learn, document and maintain.

One approach might be to structure this as two public methods:
`get_mode_displacement(qpt_index, mode, amplitude, supercell)` and
`get_combined_displacements(amplitude, supercell)` where the latter `amplitude` array runs over all q-points and indices. These would each generate an appropriate (scaled) eigenvector then call a common private method `_displaced_crystal(crystal, eigenvector, supercell)`.

Then again, the code to generate the eigenvectors is very simple. Perhaps a better API would be to add nothing at all to QpointPhononModes, and instead create a `Crystal.apply_eigenvector(eigenvector, supercell) -> Crystal` method on Crystal. That is more general and would support other (e.g. random) schemes of generating mass-scaled displacements. And to recreate these functions the user does

```
modes.crystal.apply_eigenvector(modes.eigenvectors[qpt_index, mode] * disp_amplitude, supercell)
```
or
```
weighted_evec = np.sum(modes.eigenvectors * weights, axis=(0, 1))
modes.crystal.apply_eigenvector(weighted_evec, supercell)
```

Reply:
The backstory is that I was first implemented `get_mode_displacement(qpt_index, mode, amplitude, supercell)`-like function, then I wanted to look at a double degenerated phonon. The eigenvectors of the degenerated subspace were not to my liking, and I wanted to take their linear combination with `get_combined_displacements(amplitude, supercell)`-like function. In the end I managed to get what I want, but left the function in this awkward state.

I feel like the `apply_eigenvector` function on `Crystal` is indeed a better choice. Feels more approachable and, with your snippets with the choice of the eigenvector, it is very clean.